### ISSUE TITLE
`background-processor` `futures` tests leave dangling test directories 

### ISSUE THREAD

**Issue by valentinewallace** [lightningdevkit/rust-lightning#2224]
Running `cargo test --features=futures` results in some leftover test directories from async background processor testing that should be deleted at the end of the test, via the `Drop` implementation for `Node`. It looks like `drop` is called, so not sure what's going on here.

**Comment by dunxen** [comment]
Yeah, also noticed this. I'm also changing them to use `temp_dir()` as it was bugging me lol

**Comment by dunxen** [comment]
> Is there anything left to do here? Do we know in which case(es) this is possible, or is it believe to not be?

Only happens in the async cases. I was still trying to figure out the culprit but the PR makes sure it's not going to leave the working directory dirty at least. Only that the two directories for the async test still stay behind in /tmp.

So up to you if that's ok or I should look a little further.

**Comment by valentinewallace** [comment]
It's also always been the case that if you `control+C` while background processor tests are running, test directories will not be cleaned up. Not sure if that's solvable without moving to mock-persisting rather than actually writing to disk (which seems ok, though actually writing to disk helped us catch several Windowz bugs in the past).
